### PR TITLE
Replace endpoint /data/validate with /data/estimate-fee

### PR DESCRIPTION
**Pull request by sdavidp**

### CODE REVIEW

File: modules/shared/src/main/scala/org/tessellation/currency/dataApplication/package.scala [hunk 1]
```diff
@@ -202,6 +213,12 @@ object BaseDataApplicationContextualOps {
       ): F[DataApplicationValidationErrorOr[Unit]] =
         service.validateFee(gsOrdinal)(update.asInstanceOf[Signed[D]])
 
+      override def estimateFee(gsOrdinal: SnapshotOrdinal)(update: DataUpdate)(
+        implicit context: Context,
+        A: Applicative[F]
+      ): F[EstimatedFee] =
+        service.estimateFee(gsOrdinal)(update.asInstanceOf[Signed[D]])
```
Comment (scasplte2) [inline modules/shared/src/main/scala/org/tessellation/currency/dataApplication/package.scala:220]: why cast this to a `Signed`?

File: modules/shared/src/main/scala/org/tessellation/currency/schema/EstimatedFee.scala [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+
+import org.tessellation.schema.address.Address
+import org.tessellation.schema.balance.Amount
+
+import derevo.cats.{order, show}
+import derevo.circe.magnolia.encoder
+import derevo.derive
+
+@derive(encoder, order, show)
+case class EstimatedFee(fee: Amount, address: Option[Address])
```
Comment (scasplte2) [inline modules/shared/src/main/scala/org/tessellation/currency/schema/EstimatedFee.scala:11]: I don't understand the purpose of the `Option[Address]` in the fee estimation. For the Fee itself I can see needing an `Address` and `Amount` but the use here is unclear to me.

File: modules/shared/src/main/scala/org/tessellation/currency/dataApplication/package.scala (hunk 1 above)
Comment (sdavidp) [reply modules/shared/src/main/scala/org/tessellation/currency/dataApplication/package.scala:220]: The class is defined in terms of the type variable `D <: DataUpdate`
`update` is of type `Signed[DataUpdate]` and the method requires a `Signed[D]`

File: modules/shared/src/main/scala/org/tessellation/currency/schema/EstimatedFee.scala (hunk 2 above)
Comment (marcinwadon) [reply modules/shared/src/main/scala/org/tessellation/currency/schema/EstimatedFee.scala:11]: @sdavidp If you say so, it should be either no fee or fee + address. Right now you allow an impossible state to be defined: fee + no address.

So it should be defined as below:
```
sealed trait EstimatedFee

case object Zero extends EstimatedFee
case class Estimated(amount: Amount, address: Address) extends EstimatedFee

```

Comment (sdavidp) [reply modules/shared/src/main/scala/org/tessellation/currency/schema/EstimatedFee.scala:11]: @marcinwadon Ok, I will change the implementation. However, the ticket wants the encoded JSON for _zero_ to be `{ "fee" : 0 }`

File: modules/shared/src/main/scala/org/tessellation/currency/dataApplication/package.scala (hunk 1 above)
Comment (sdavidp) [reply modules/shared/src/main/scala/org/tessellation/currency/dataApplication/package.scala:220]: As discussed in our meeting, the method does NOT want a `Signed[D]`. I'll fix it.

File: modules/shared/src/main/scala/org/tessellation/currency/schema/EstimatedFee.scala (hunk 2 above)
Comment (sdavidp) [reply modules/shared/src/main/scala/org/tessellation/currency/schema/EstimatedFee.scala]: @marcinwadon Updated.

**Comment by marcinwadon** [comment]
@sdavidp Hey, there is a label missing

**Review by scasplte2 — APPROVED** [review]
everything seems in order here now once the base is updated

**Review by marcinwadon — APPROVED** [review]

**Pull request merged**